src/csharp/Microsoft.Spark/RDD.cs

```diff
@@ -298,24 +298,12 @@ public IEnumerable<T> Collect()
         private (int, string) CollectAndServe()
         {
             JvmObjectReference rddRef = GetJvmRef();
-            object result = rddRef.Jvm.CallStaticJavaMethod(
+            var pair = (JvmObjectReference[])rddRef.Jvm.CallStaticJavaMethod(
                 "org.apache.spark.api.python.PythonRDD",
                 "collectAndServe",
                 rddRef.Invoke("rdd"));
 
-            if (result is int @port)
-            {
-                // In spark 2.3.0, collectToPython() returns a port number.
-                return (@port, string.Empty);
-            }
-            else
-            {
-                // From spark >= 2.3.1, collectToPython() returns a pair
-                // where the first is a port number and the second is the secret
-                // string to use for the authentication.
```

Comment:
I think we can leave this comment (without "From spark >= 2.3.1)

Reply:
Readded comment.